### PR TITLE
[FXML-4647] Tests for Torch to Linalg backend pipeline

### PR DESCRIPTION
**Pull request by cferry-AMD**
These tests check the pass options `use-mlprogram` and `verify` added in [this commit](https://github.com/Xilinx/torch-mlir/commit/eea43ececf812035c669f5c1c35fa81d5f94c929) work correctly.

### CODE REVIEW

File: test/Dialect/TorchConversion/torch-backend-to-linalg-on-tensors-no-mlprogram.mlir [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// RUN: torch-mlir-opt -p 'builtin.module(torch-backend-to-linalg-on-tensors-backend-pipeline{use-mlprogram=0})' -split-input-file %s | FileCheck %s
```
Comment (TinaAMD) [inline test/Dialect/TorchConversion/torch-backend-to-linalg-on-tensors-no-mlprogram.mlir:1]: Maybe add the same test with `use-mlprogram=1` and a check for the global?

File: test/Dialect/TorchConversion/torch-backend-to-linalg-on-tensors-no-contract-check.mlir [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// RUN: torch-mlir-opt -p 'builtin.module(torch-backend-to-linalg-on-tensors-backend-pipeline{verify=0})' -split-input-file %s | FileCheck %s
```
Comment (TinaAMD) [inline test/Dialect/TorchConversion/torch-backend-to-linalg-on-tensors-no-contract-check.mlir:1]: I guess the verification already has some tests?

File: test/Dialect/TorchConversion/torch-backend-to-linalg-on-tensors-no-mlprogram.mlir (hunk 1 above)
Comment (cferry-AMD) [reply test/Dialect/TorchConversion/torch-backend-to-linalg-on-tensors-no-mlprogram.mlir:1]: The pass that adds the global is tested in `test/Conversion/TorchConversionToMLProgram/basic.mlir`; then the test that checks the pipeline altogether doesn't check individual components (including this global). Does it make sense to have a test for this individual component?

File: test/Dialect/TorchConversion/torch-backend-to-linalg-on-tensors-no-contract-check.mlir (hunk 2 above)
Comment (cferry-AMD) [reply test/Dialect/TorchConversion/torch-backend-to-linalg-on-tensors-no-contract-check.mlir:1]: Yes, the neighboring `verify-linalg-on-tensors-backend-contract.mlir` file takes care of this.

File: test/Dialect/TorchConversion/torch-backend-to-linalg-on-tensors-no-mlprogram.mlir (hunk 1 above)
Comment (TinaAMD) [reply test/Dialect/TorchConversion/torch-backend-to-linalg-on-tensors-no-mlprogram.mlir:1]: I don't mean to really check the global, but the flag working as expected (kind of what the test setting =0 does). Does that make it more clear?

File: test/Dialect/TorchConversion/torch-backend-to-linalg-on-tensors-no-mlprogram.mlir [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+// CHECK-NOT: ml_program.global private mutable @global_seed(dense<0> : tensor<i64>) : tensor<i64>
```
Comment (josel-amd) [inline test/Dialect/TorchConversion/torch-backend-to-linalg-on-tensors-no-mlprogram.mlir]: Nit: Is it possible to only check that there are no references to the `ml_program` dialect? This kind of test can be very easy to break.

File: test/Dialect/TorchConversion/torch-backend-to-linalg-on-tensors-no-mlprogram.mlir (hunk 1 above)
Comment (cferry-AMD) [reply test/Dialect/TorchConversion/torch-backend-to-linalg-on-tensors-no-mlprogram.mlir:1]: I added a version with that flag being enabled :)

**Review by TinaAMD — APPROVED** [review]
LGTM

**Pull request merged**